### ISSUE TITLE
check_subrc, skip check if assigning and checking assigning

### ISSUE THREAD

**Issue by larshp** [abaplint/abaplint#1669]
should not report error for

```abap
READ TABLE ct_failed_objects ASSIGNING <lfs_failed_objects> INDEX 1.
IF <lfs_failed_objects> IS ASSIGNED.
ENDIF.
```

**Comment by victorizbitskiy** [comment]
Hi Lars!

I noticed that there is still a sy-subrc check for the code:
```abap
DATA: lcl_ref TYPE REF TO data.
FIELD-SYMBOLS: <restab_standatd> TYPE STANDARD TABLE,

ASSIGN: lcl_ref->* TO <restab_standatd>.
IF <restab_standatd> IS NOT ASSIGNED.
ENDIF.
```

**Comment by larshp** [comment]
@victorizbitskiy thanks, please open a new issue for the above

**Comment by victorizbitskiy** [comment]
@larshp done. 

https://github.com/abaplint/abaplint/issues/1863
